app/controllers/withdraw_controller.rb

```diff
@@ -35,10 +35,16 @@ def create
 
 private
 
-  def check_permission
-    return if current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
-
+  def check_permissions
     @edition = Edition.find_current(document: params[:document])
-    render :non_managing_editor, status: :forbidden
+
+    unless current_user.can_access_in_history_mode?(@edition) && current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
```

Comment:
I've found myself a bit confused by this conditional not sure where the managing editor part kicks in. Might need you to explain this to me.

Reply:
Discussed offline and now sorted!